types/which/index.d.ts

```diff
@@ -1,47 +1,57 @@
 /** Finds all instances of a specified executable in the PATH environment variable */
-declare function which(cmd: string, options: which.Options & which.AsyncOptions & which.OptionsAll): Promise<string[]>;
-declare function which(cmd: string, options?: which.Options & which.AsyncOptions & which.OptionsFirst): Promise<string>;
+
+type AddNull<TOptions, TRet> = TOptions extends { nothrow: infer TVal }
+    // nothrow is specified
+    ? TVal extends false
+        // TVal is false
+        ? TRet
+        // TVal is boolean or true
+    : TRet | null
+    // nothrow not specified
+    : TRet;
+
+type AddArray<TOptions, TRet> = TOptions extends { all: infer TVal }
```

Comment:
Can you expand on the naming for AddNull and AdddArray? I see that the new type a bit obscure to understand what it represents.

When i see the code feels that you are more append or transforming the original types

Reply:
Sure, happy to do that. What name would work best? Maybe NullableIfNoThrow and ArrayIfAll ?

Reply:
> Sure, happy to do that. What name would work best? Maybe NullableIfNoThrow and ArrayIfAll ?

I was thinking something quite similar. Something in lines`AppendNullIfNothrow` and `TransformToArrayIfAll`. What are your thoughts? If you prefer yours, i will accept them as well

Reply:
Sounds good, will amend later.

Reply:
Done!

Reply:
Well done, i appreciated the improvements